src/hooks.ts

```diff
@@ -1,13 +1,35 @@
+import React from 'react';
 import AsyncStorage from './AsyncStorage';
-import type { AsyncStorageHook } from './types';
 
-export function useAsyncStorage(key: string): AsyncStorageHook {
-  return {
-    getItem: (...args) => AsyncStorage.getItem(key, ...args),
-    setItem: (...args) => AsyncStorage.setItem(key, ...args),
-    mergeItem: (...args) =>
+export function useAsyncStorage(key: string) {
+  const getItem = React.useCallback(
+    (...args) => AsyncStorage.getItem(key, ...args),
+    [key]
+  );
+
+  const setItem = React.useCallback(
+    //@ts-ignore
+    (...args) => AsyncStorage.setItem(key, ...args),
+    [key]
```

Comment:
Do we have to create a new array for each of the callbacks? Can we create it once and reuse?

Reply:
No, this doesn't work. Eslint will complain:
```tsx
const dep = [key]
const getItem = React.useCallback(
  (...args) => AsyncStorage.getItem(key, ...args),
  dep
);
```

```
const dep: string[]

React Hook React.useCallback was passed a dependency list that is not an array literal. This means we can't statically verify whether you've passed the correct dependencies.eslint[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)
React Hook React.useCallback has a missing dependency: 'key'. Either include it or remove the dependency array.eslint[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)
```